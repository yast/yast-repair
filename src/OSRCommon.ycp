/**
 *  File:
 *	OSRCommon.ycp
 *
 *  Module:
 *	repair
 *
 *  Summary:
 *	Common functions and variables, used by other OSR modules
 *
 *  Author:
 *	Jiri Suchomel <jsuchome@suse.de>
 *
 * $Id$
 */
{

/**
 * Variables and functions in these module were moved from other
 * OSR* modules to break their cyclic dependencies.
 *
 * This module should not import any other OSR module
 */

module "OSRCommon";

textdomain "repair";

/**
 * Functions that could be used as a default in lookups for functin pointers
 * @return false
 */
global define boolean False () ``{
    return false;
}

/**
 * @return `error
 */
global define symbol SymbolError () ``{
    y2error ("method not found");
    return `error;
}

/**
 * @return ""
 */
global define string EmptyString () ``{
    return "";
}

/**
 * @return $[]
 */
global define map EmptyMap () ``{
    return $[];
}

/**
 * @return []
 */
global define list EmptyList () ``{
    return [];
}


// ---------------------------------------------------------------------
// from OSRDirect.ycp:

/**
 * The name of the current direct repair method.
 */
global string current_direct_name	= "";


// ---------------------------------------------------------------------
// from OSRRepair.ycp:

/**
 * The name of the repair and rescue tool.
 */
global string tool_name		= _("System Repair");


// ---------------------------------------------------------------------
// from OSR.ycp:

/**
 * The name of the current module during detection.
 * e.g.: osr_module_partition
 */
global string current_module_name	= "";


/**
 * ["init", "mbr_check", "swap_check", "fs_check", "fstab_check"]
 */
global list<string> detect_group_list	= [];

/**
 * The current detection map during detection.
 *
 * $[
 "group":"init",
 "method": ``OSRInitFloppy(),
 "progress":10,
 "provides":["repair_target"],
 "requires":[],
 "summary":"Init target system"
 ]
 */
global map current_detect_map	        = $[];

/**
 * $[	"fs_check" : $[
 *		"help"	: "Mounting a filesystem is not possible",
 *		"text"	: "Check filesystem"
 *	],
 *	"fstab_check" : $[
 *		"help"	: "Mounting a filesystem is not possible",
 *		"text"	: "Check fstab entries"
 *	],
 *	"init" : $[
 *		"help"	: "",
 *		"mode"	: "forall",
 *		"text"	: "Init repair system"
 *	],
 *	"mbr_check" : $[
 *		"help"	: "If you have installed a other os after inst. Linux",
 *		"text"	: "Check master boot record"
 *	],
 *	........
 * ]
 */
global map global_entries		= $[];

  /**
   * The global_provides map contains entries created during the system scan.
   * The different detection methods access to the global_provides map to get
   * informations from former detection methods.
   * $["fs_module_loaded":true,
       "has_floppy":true,
       "just_mounted":[$["mountpoint":"/mnt", "partition":"/dev/hda3"]],
       "linux_partition_list":["/dev/hda1", "/dev/hda3", "/dev/hda6"],
       "mount_possible_list":["/dev/hda1", "/dev/hda3", "/dev/hda6"],
       "mounted_partitions":[$["file":"/mnt", "freq":0, "mntops":"rw", "passno":0, "spec":"/dev/hda3", "vfstype":"reiserfs"],..,],
       "repair_target":"",
       "root_mounted":true,
       "root_mountpoint":"/mnt",
       "root_partition":"/dev/hda3",
       "root_partition_type":"reiserfs",
       "swap_partition_list":["/dev/hda2", "/dev/hda5"],
       "swapable_partitions":["/dev/hda2", "/dev/hda5"],
       "valid_fstab":true,
       "valid_root_partitions":["/dev/hda3"],
       "valid_swap_partitions":["/dev/hda2", "/dev/hda5"],
       "valid_target":true, "valid_target_list":["/dev/hda"]]
   */
  map<string,any> global_provides			= $[];


  /**
   *  Stores the given list with the specified name into the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the provided value.
   *  @param list value The provided value.
   *  @return boolean True.
   */
  global define boolean ProvideList(string name, list value) ``{
      global_provides = add (global_provides, name, value );
      return true;
  };

  /**
   *  Stores the given map with the specified name into the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the provided value.
   *  @param map value The provided value.
   *  @return boolean True.
   */
  global define boolean ProvideMap(string name, map value) ``{
      global_provides	= add(global_provides, name, value );
      return true;
  };

  /**
   *  Stores the given string with the specified name into the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the provided value.
   *  @param string value The provided value.
   *  @return boolean True.
   */
  global define boolean ProvideString(string name, string value) ``{
      global_provides	= add(global_provides, name,value ); // provides_map  );
      return true;
  };

  /**
   *  Stores the given path with the specified name into the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the provided value.
   *  @param path value The provided value.
   *  @return boolean True.
   */
  global define boolean ProvidePath(string name, path value) ``{
      global_provides	= add(global_provides, name, value ); //provides_map  );
      return true;
  };

  /**
   *  Stores the given boolean value with the specified name into the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the provided value.
   *  @param boolean value The provided value.
   *  @return boolean True.
   */
  global define boolean ProvideBoolean(string name, boolean value) ``{
      global_provides	= add(global_provides, name,value); // provides_map  );
      return true;
  };

  /**
   *  Returns the required string from the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the required value.
   *  @return string value The required string, "" as default.
   */
  global define string RequireString(string name) ``{

      if ( ! haskey( global_provides, name ) )
      {
	  y2warning ("The required value \"%1\" is not yet set.", name);
      }

      return global_provides[ name ]:"";
  };

  /**
   *  Returns the required list from the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the required value.
   *  @return string The required list, [] as default.
   */
  global define list RequireList(string name) ``{

      if (! haskey (global_provides, name))
      {
	  y2warning ("The required value \"%1\" is not yet set.", name);
      }

      return global_provides[name]:[];
  };

  /**
   *  Returns the required map from the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the required value.
   *  @return map The required map, $[] as default.
   */
  global define map RequireMap(string name) ``{

      if ( ! haskey ( global_provides, name ))
      {
	  y2warning ("The required value \"%1\" is not yet set.", name);
      }

      return global_provides[ name]:$[];
  };

  /**
   *  Returns the required path from the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the required value.
   *  @return path value The required path, . as default.
   */
  global define path RequirePath(string name) ``{

      if ( ! haskey( global_provides , name ))
      {
	  y2warning ("The required value \"%1\" is not yet set.", name);
      }

      return global_provides[name]:.;
  };

  /**
   *  Returns the required boolean value from the global_provides map
   *  in the osr_map.
   *
   *  API function,
   *
   *  @param string name The name of the required value.
   *  @return boolean value The required boolean value, false as default.
   */
  global define boolean RequireBoolean(string name) ``{

      if (! haskey ( global_provides, name ))
      {
	  y2warning ("The required value \"%1\" is not yet set.", name);
      }

      return global_provides[ name]:false;
  };



  /**
   * Reset variables moved from OSR.ycp
   */
  global define void Reset () ``{

      global_provides	= $[];

      current_module_name	= "";
      current_detect_map	= $[];

      global_entries		= $[];
      detect_group_list	= [];
  }

  /**
   * Set the new value of global_provides map
   */
  global define void SetGlobalProvides (map new) ``{

      global_provides	= eval ((map<string,any>) new);
  }

  /**
   * Set the new value of global_provides map
   */
  global define map<string,any> GlobalProvides () ``{

      return global_provides;
  }

/**
 * Get group data.
 */
global define map GetGroupMap (string key) ``{

    return global_entries[key]:$[];
}

  /**
   *  Checks if the first specified list (of strings) is a subset of
   *  the second one. The sequence of entries in the lists is nonrelevant.
   *
   *  For internal use only.
   *  See the testsuite.
   *
   *  @example OSRIsSubSet([], ["a", "b"]) -> true;
   *  @example OSRIsSubSet(["a"], ["a", "b"]) -> true;
   *  @example OSRIsSubSet(["c"], ["a", "b"]) -> false;
   *  @example OSRIsSubSet(["a", "b", "c"], ["a", "b"]) -> false;
   *  @param list The first list.
   *  @param list The second list.
   *  @return True if all elements of l1 are contained in l2 or if l1 is empty.
   */
  global define boolean IsSubSet (list<string> l1, list<string> l2) {

      integer index = 0;

      // remove duplicate entries and sort the lists
      l1 = toset(l1);
      l2 = toset(l2);

      // check special cases
      if ((size(l1) < 0) || (size(l2) < 0))   return false;

      else if (size(l1) > size(l2))  return false;

      else if ((size(l1) == 0)) return true;

      else if (size(l1) <= size(l2))
      {
	  // this is the main subset-checker
	  while ((index >= 0) && (index < size(l1)))
	  {
	      if (!contains (l2, l1[index]:""))
	      {
		  return false;
	      }
	      index = index + 1;
	  }
	  return true;
      }
      else
      {
	  return false;
      }
  };

}//EOF
