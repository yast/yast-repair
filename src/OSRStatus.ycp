/**
 *  File:
 *   OSRStatus.ycp
 *
 *  Module:
 *   YaST2 OS Repair
 *
 *  Summary:
 *
 *
 *  Authors:
 *   Johannes Buchhold <jbuch@suse.de>, 2002
 *
 * $Id$
 */
{
  module "OSRStatus";

  textdomain "repair";

  global string status				= "";
  global string severity			= "";
  global string next_detect_method_name		= "";


  map status_map	= $[
			    "detect_ok" 	: $[
						    "color" 	  :  "blue",
						    "status_text" :  _("Ok"),
						    "priority"	  : 3,
						    "type"	  : "detect"
			    ],
			    "detect_error" 	: $[
						    "color" 	  :	"red",
						    "status_text" :  _("error"),
						    "priority"    : 6,
						    "type"	  : "detect",
			    ],
			    "detect_problem" 	: $[
						    "color" 	  : "red",
						    "status_text" :  _("problematic"),
						    "type"	  : "detect",
						    "priority"    : 5
			    ],
			    "detect_omit" 	: $[
						    "color" 	  : "blue",
						    "status_text" :  _("omitted"),
						    "type"	  : "detect",
						    "priority"    : 4
			    ],
			    "repair_ok" 	: $[
						    "color" 	  : "green",
						    "status_text" : _("repaired"),
						    "type"	  : "repair",
						    "priority"    : 7
			    ],
			    "repair_error" 	: $[
						    "color" 	     : "red",
						    "status_text"    : _("failed"),
						    "type"	     : "repair",
						    "priority"       : 9
			    ],
			    "repair_omit" 	: $[
						    "color"  	      : "red",
						    "status_text"     : _("omitted"),
						    "type"	      : "repair",
						    "priority"        : 8
			    ],
			    "open"	 	: $[
						    "color"		: "green",
						    "status_text"	: _("open"),
						    "type"	        : "detect",
						    "priority"		: 1
			    ],
			    "not_found"		: $[
						    "color"		: "blue",
						    "status_text"	: _("not found"),
						    "type"	  	: "detect",
						    "priority"		: 2
						    ]
  ];


  global define void RestartScan(string tnext_detect_method_name)``{
      next_detect_method_name =tnext_detect_method_name;
  }

  global define void Reset()``{
      status = "";
      severity = "";
      next_detect_method_name = "";
  }

  global define string Color(string local_status) ``{
      return status_map [ local_status, "color"]:"blue";
  }


  global define string Text(string local_status) ``{
      return status_map [ local_status, "status_text"]:"";
  }

  global define integer Priority(string local_status) ``{
      return status_map [ local_status, "priority"]:1;
  }

  global define string PriorityStr(integer priority ) ``{

      list status_list = maplist (string key, map val, (map<string,map<any,any> >) status_map,``(val));
      map pr_map = (map<string,any>) find (map<string,any> v, (list<map<string,any> >)status_list,
	``( v["priority"]:1 == priority ));
      if ( pr_map != nil && pr_map != $[] )
	  return pr_map["status_text"]:"detect_ok";
      else return "detect_ok";
  }

  global define string PriorityKey(integer priority ) ``{

      list status_list = maplist (string key, map val,(map<string,map<any,any> >) status_map,
	``( add (val, "key", key ) ));
      map pr_map = (map<string,any>) find (map<string,any> v, (list<map<string,any> >)status_list,
	``( v["priority"]:1 == priority ));
      if ( pr_map != nil && pr_map != $[] )
	  return pr_map["key"]:"";
      else return "detect_ok";
  }

  /**
   *  Returns the level of severity of the detected error.
   *  This severity involves the rescue system to stop the further execution if
   *  the detected error is not repaired. E.g. if the rescue system does not
   *  detect any harddisks it makes no sense to go on checking for any errors.
   *  The rescue system will halt.
   *
   *  API function.
   *
   *  @return string The severity of the detected error.
   */
  global define boolean IsErrorSeverityBlocking() ``{
      return ( severity == "blocking");
  };

  /**
   *  Returns the level of severity of the detected error. This severity means
   *  that the detected error involves to jump to the next detection module
   *  without executing any further method of the current module. E.g. if the
   *  rescue system detects that the LILO package is not installed on the system
   *  and this error could not be repaired, it makes no sense to check the LILO
   *  configuration file for any errors. So the rescue system jumps to the next
   *  module in the module sequence.
   *
   *  API function.
   *
   *  @return string The severity of the detected error.
   */
  global define boolean IsErrorSeverityModuleBlocking() ``{
      return ( severity == "module_blocking");
  };

  /**
   *  Sets the level of severity of the detected error.
   *  API function.
   *  @return string The severity of the detected error.
   */
  global define void ErrorSeverityBlocking() ``{
      severity = "blocking";
  };

  /**
   *  Sets the level of severity of the detected error.
   *  API function.
   *  @return string The severity of the detected error.
   */
  global define void ErrorSeverityModuleBlocking() ``{
      severity = "module_blocking";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   */
  global define void DetectOK2Name(string tnext_detect_method_name) ``{
      status =  "detect_ok";
      severity = "";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   */
  global define void DetectError2Name(string tnext_detect_method_name) ``{
      status = "detect_error";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define void DetectProblem2Name(string tnext_detect_method_name) ``{
      status = "detect_problem";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define void RepairOK2Name(string tnext_detect_method_name) ``{
      status   = "repair_ok";
      severity = "";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define void RepairError2Name(string tnext_detect_method_name) ``{
      status = "repair_error";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define void RepairOmit2Name(string tnext_detect_method_name) ``{
      status = "repair_omit";
      next_detect_method_name = tnext_detect_method_name;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define void NotFound2Name(string tnext_detect_method_name) ``{
      status = "not_found";
      next_detect_method_name = tnext_detect_method_name;
  };

  global define void NotFound()``{
      status   = "not_found";
      severity = "";
      next_detect_method_name = "";
  }

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void DetectOK() ``{
      status =  "detect_ok";
      severity = "";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void DetectError() ``{
      status = "detect_error";
      next_detect_method_name = "";
  };


  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void DetectOmit() ``{
      status = "detect_omit";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void DetectProblem() ``{
      status = "detect_problem";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void RepairOK() ``{
      status   = "repair_ok";
      severity = "";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void RepairError() ``{
      status = "repair_error";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void RepairOmit() ``{
      status = "repair_omit";
      next_detect_method_name = "";
  };

  global define void Open() ``{
      status   = "open";
      severity = "";
      next_detect_method_name = "";
  }

  global define void Cancel()``{
      status = "cancel";
      severity = "";
      next_detect_method_name = "";
  }

 /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   */
  global define void Exit() ``{
      status = "exit";
      next_detect_method_name = "";
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Is OK?
   */
  global define boolean IsStatusDetectOK() ``{
      return ( status ==  "detect_ok");
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Is error?
   */
  global define boolean IsStatusDetectError() ``{
      return ( status == "detect_error");
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Omitted?
   */
  global define boolean IsStatusDetectOmit() ``{
      return ( status == "detect_omit");
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Is detect problem?
   */
  global define boolean IsStatusDetectProblem() ``{
      return ( status == "detect_problem") ;
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Repaired?
   */
  global define boolean IsStatusRepairOK() ``{
      return ( status == "repair_ok");
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Repair failed?
   */
  global define boolean IsStatusRepairError() ``{
      return ( status == "repair_error");
  };

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return Repait omitted?
   */
  global define boolean IsStatusRepairOmit() ``{
      return ( status == "repair_omit");
  };

  global define boolean IsStatusNotFound()``{
      return ( status == "not_found");
  }

  /**
   *  This function is a wrapper for the definition of the correct status.
   *
   *  API function,
   *
   *  @return void
   */
  global define boolean IsStatusExit() ``{
      return ( status == "exit");
  };

  global define boolean IsOpen()``{
      return ( status == "open");
  }

  global define boolean IsCancel()``{
      return ( status == "cancel");
  }

}
