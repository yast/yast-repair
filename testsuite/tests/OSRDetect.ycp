/**
 *  Test for running automatic detection
 *  Author:	Jiri Suchomel <jsuchome@suse.cz>
 *  $Id$
 */

{
    // testedfiles: OSRRepairUI.ycp OSR.ycp OSRCommon.ycp osr_module_partition.ycp OSRFstab.ycp

    import "Testsuite";

    map I_READ = $[
	"probe" : $[
	    "cdrom"	: [],
	    "is_uml"	: false,
	    "floppy"	: $[
		"manual"	: [ $[] ] // 1 floppy for "init_floppy" method
	    ],
	    "architecture"	: ""
	],
	"etc"	: $[
	    "mtab"	: []
	],
	"proc"	: $[
	    "modules"	: $[],
	    "swaps"	: [],
	    "mounts"	: [ $["file": "/" , "spec": "/dev/hda3"]],
	],
	"target" : $[
	    "size"	: -1,
	    "tmpdir"	: "/tmp/",
	],
	"product"	: $[
	    "features"	: $[
		"USE_DESKTOP_SCHEDULER"	: "",
		"IO_SCHEDULER"		: "",
		"UI_MODE"		: "",
		"ENABLE_AUTOLOGIN"	: "",
		"EVMS_CONFIG"		: ""
	    ],
	],
    ];

    map I_EXEC = $[
	"target": $[
	    "bash_output" : $[],
	],
    ];
    Testsuite::Init ( [I_READ, $[], I_EXEC], nil);

    import "Mode";

    Mode::SetTest ("testsuite");

    import "OSR";
    import "OSRCommon";
    import "OSRStatus";

    map READ	= $[
	"target" : $[
	    "yast2"	: $[],
	    "size"	: 1,
	    "string"	: "/dev/hda3 / reiserfs acl,user_xattr 1 1",//fstab
	    "tmpdir"	: "/tmp/",
	    "ycp"	: $[],
	],
	"md"	: $[
	    "config"	: $[],
	],
	"lvm"	: $[
	    "vg"	: $[],
	    "pv"	: $[],
	],
	"evms"	: $[
	    "container"	: [],
	    "volume"	: [],
	],
	"probe" : $[
	    "boot_disk"	: "",
	    "disk"	: [],
	    "floppy"	: [ $[] ],

	    "cdrom"	: [],
	    "is_uml"	: false,
	    "floppy"	: $[
		"manual"	: [ $[] ] // 1 floppy for "init_floppy" method
	    ],
	    "architecture"	: ""
	],
	"proc"	: $[
	    "swaps"	: [ ],
	    "mounts"	: [ $["file": "/" , "spec": "/dev/hda3"],
			    $["file": "/mnt" , "spec": "/dev/hda3"] ],
	    "modules"	: $[],
	],
	"etc"	: $[
	    "mtab"	: [ $["file": "/"] ],
	    "install_inf"	: $[
		"Cdrom"	: "",
	    ],
	],
	"sysconfig"	: $[
	    "kernel"	: $[
		"INITRD_MODULES"	: "",
	    ],
	    "bootloader": $[
		"LOADER_TYPE"		: "grub",
	    ],
	    "language"	: $[
		"RC_LANG"		: "",
		"DEFAULT_LANGUAGE"	: "",
		"ROOT_USES_LANG"	: "",
		"LANGUAGES"		: "",
	    ],
	],
	"xml"		: $[],
	"grub"		: [],
	"bootloader"	: $[
	    "settings"	: $[],
	],
    ];
    map WRITE	= $[];
    map EXEC	= $[
	"target": $[
	    "bash_output" : $[],
	],
    ];

    Testsuite::Dump (sformat ("global entries: %1", OSRCommon::global_entries));
    Testsuite::Test (``(OSR::Init ()), [READ, WRITE, EXEC], 0);
    Testsuite::Dump (sformat ("global entries (keys): %1", maplist (
	string k, map m, (map<string,map>)OSRCommon::global_entries, ``(k))));

    Testsuite::Dump (sformat ("detect group: %1", OSR::DetectGroupList ()));

    Testsuite::Test (``(OSR::SelectDefaultDetectGroups ()), [READ, WRITE, EXEC], 0);

    Testsuite::Dump (sformat ("detect group: %1", OSR::DetectGroupList ()));

    Testsuite::Dump ("----------------- start detection [no hard disk found]: ------");

    Testsuite::Test (``(OSR::Detect ()), [READ, WRITE, EXEC], 0);

    Testsuite::Dump ("--------------------------------------------------------------");

    Testsuite::Dump (sformat ("--- status: %1", OSRStatus::status));

    Testsuite::Dump (sformat ("--- failed in module: %1", OSRCommon::current_module_name));

    Testsuite::Dump ("----------------- start detection [hard disks now ok]: --------");

    READ["target","yast2"]	= $[
	"/dev/hda": $[
	    "device"	: "/dev/hda",
	    "type"      : `CT_DISK,
	    "partitions": [
		$[
		    "detected_fs":`reiser,
		    "device"	: "/dev/hda3",
		    "mount"	: "/",
		    "nr"	: 3
		],
		$[
		    "fsid" : 130,
		    "device"	: "/dev/hda1",
		    "nr"	: 1
		]
	    ]
	]
    ];

    Testsuite::Test (``(OSR::Detect ()), [READ, WRITE, EXEC], 0);
}
