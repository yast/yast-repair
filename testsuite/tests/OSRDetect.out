Dump	global entries: $[]
Return	true
Dump	global entries (keys): ["boot_check", "fs_check", "fstab_check", "init", "mbr_check", "packages_db_check", "packages_selection", "packages_verify_all", "packages_verify_base", "swap_check"]
Dump	detect group: []
Return	nil
Dump	detect group: ["init", "mbr_check", "swap_check", "fs_check", "fstab_check", "packages_db_check", "packages_selection", "packages_verify_base", "boot_check"]
Dump	----------------- start detection [no hard disk found]: ------
Return	`error
Dump	--------------------------------------------------------------
Dump	--- status: detect_error
Dump	--- failed in module: osr_module_partition
Dump	----------------- start detection [hard disks now ok]: --------
Read	.etc.mtab [$["file":"/"]]
Log	Skip detection method (requires not solved ) mount_all
Log	Skip detection method (requires not solved ) find_package_db
Log	Skip detection method (requires not solved ) find_minimal_selection
Log	Skip detection method (requires not solved ) verify_base_packages
Log	Skip detection method (requires not solved ) find_config_files
Log	Skip detection method (requires not solved ) find_bootloader_type
Log	Skip detection method (requires not solved ) find_boot_root
Log	Skip detection method (requires not solved ) bootloader_package_installed
Log	Skip detection method (requires not solved ) verify_bootloader_package
Log	Skip detection method (requires not solved ) bootloader_package_version
Log	Skip detection method (requires not solved ) bootloader_initrd_modules
Log	Skip detection method (requires not solved ) bootloader_conf_exists
Log	Skip detection method (requires not solved ) bootloader_conf_check
Log	Skip detection method (requires not solved ) bootloader_ask_user
Return	`detectok
